src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -5235,5 +5236,214 @@ void test(SemanticModelOptions options)
                 Assert.Equal(PublicNullableAnnotation.None, typeInfo.Type.NullableAnnotation);
             }
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71522")]
+        public void CollectionExpression_NestedNullability_01()
+        {
+            var source = """
+                #nullable enable
+
+                var b = false;
+                var arr = b ? ["1"] : new[] { "2" };
+                """;
+
+            var comp = CreateCompilation(source);
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var root = tree.GetRoot();
+            var collectionExpr = root.DescendantNodes().OfType<CollectionExpressionSyntax>().Single();
```

Comment:
nit: just FYI in case you find it convenient, I added a test helper "GetSyntax" which allows `GetSyntax<CollectionExpressionSyntax>(tree, """["1"]""")` to do the trick in this case.